src/devices/src/virtio/vsock/device.rs

```diff
@@ -131,10 +128,15 @@ where
     /// otherwise.
     pub fn process_rx(&mut self) -> bool {
         debug!("vsock: process_rx()");
+        let mem = match self.device_state {
```

Comment:
I think it was more efficient how it was written before with device_state being a simple bool. Processing the event is on the critical path and it is more efficient to just check a flag than to do this match to get the memory.

Any reason for having the memory in the enum?

Reply:
The previous version of the code had a single check of the `is_active: bool` per event handle while the code in this PR has one or sometimes two (like the case in this line) checks of the `device_state: enum DeviceState` per event handle.

`match enum { ... }` and `if boolean { ... }` are equivalent in terms of efficiency (it's still compare and jump instructions).

The extra `if` per event handle is inconsequential to this code's performance where the amount of work done for an event is substantial.

Even so, I am open to alternatives. The problem trying to be solved here is allowing device creation without memory and having memory given to the device some time later (during activation for example). Better ideas to support that are welcome! 😃

Reply:
> Any reason for having the memory in the enum?

See this https://github.com/firecracker-microvm/firecracker/pull/1687#discussion_r397885208 for a more detailed explanation.